docs/core-concepts/ui-renderers.md

```diff
@@ -104,6 +104,9 @@ export default {
   snowpackPlugin: '@snowpack/plugin-xxx', // optional, the name of a snowpack plugin to inject
   snowpackPluginOptions: { example: true }, // optional, any options to be forwarded to the snowpack plugin
   knownEntrypoint: ['framework'], // optional, entrypoint modules that will be used by compiled source
+  external: ['dep'] // optional, dependencies that should not be built by snowpack
+  polyfills: ['./shadow-dom-polyfill.js'] // optional, module scripts that should be loaded before client hydration.
+  hydrationMethod: 'self' // optional, specifies that the `client` script needs greater control over hydration.
```

Comment:
Yeah those still all work. The renderer isn't responsible for that part of it, it's responsible for loading the element. This is so you can add props to an element before the client script is loaded. This is because elements render themselves, so if the props aren't added the renderer can add them first.

This is somewhat of an edge case since most of the time element "props" are on the element as attributes. So if this is confusing I think we can probably remove that part.

Reply:
Ah that makes total sense! Any thoughts on less confusing terminology? This seems extremely useful for custom elements.

Reply:
I think I might just remove this for now and make the `client` config optional. I think it's probably premature to design around this constraint as it's fairly rare to server render an element that doesn't serialize to attributes any ways.

Reply:
I removed it, we can add back later.